### ISSUE TITLE
CSV export handling for drafts

### ISSUE THREAD

**Issue by fbjohn** [woocommerce/woocommerce#17722]
## Prerequisites

- [ x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [? ] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [x ] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. Click products
2. Click Export
3. Click Generate CSV
(With some products not being published.)

I have no idea what a master branch on github is, nor how to test it.

## Expected/actual behavior

When I follow those steps, I see...

Only published products.

I was expecting to see...

All products

## Isolating the problem

- [ x] This bug happens with only WooCommerce plugin active
- [ x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [ x] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>
`
### WordPress Environment ###

Home URL: http://fbknives.com
Site URL: http://fbknives.com
WC Version: 3.2.3
Log Directory Writable: ✔
WP Version: 4.9
WP Multisite: –
WP Memory Limit: 368 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache
PHP Version: 7.0.25
PHP Post Max Size: 128 MB
PHP Time Limit: 300
PHP Max Input Vars: 1000
cURL Version: 7.54.0
OpenSSL/1.0.2k

SUHOSIN Installed: –
Max Upload Size: 128 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.2.3
WC Database Prefix: wpqj_
MaxMind GeoIP Database: ❌ The MaxMind GeoIP Database does not exist - Geolocation will not function. You can download and install it manually from http://dev.maxmind.com/geoip/legacy/geolite/ to the path: . Scroll down to "Downloads" and download the "Binary / gzip" file next to "GeoLite Country". Please remember to uncompress GeoIP.dat.gz and upload the GeoIP.dat file only.
Total Database Size: 5.12MB
Database Data Size: 4.08MB
Database Index Size: 1.04MB
wpqj_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB
wpqj_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB
wpqj_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB
wpqj_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.05MB
wpqj_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB
wpqj_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB
wpqj_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB
wpqj_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB
wpqj_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB
wpqj_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB
wpqj_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB
wpqj_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB
wpqj_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB
wpqj_woocommerce_log: Data: 0.02MB + Index: 0.02MB
wpqj_commentmeta: Data: 0.02MB + Index: 0.03MB
wpqj_comments: Data: 0.02MB + Index: 0.08MB
wpqj_links: Data: 0.02MB + Index: 0.02MB
wpqj_options: Data: 3.02MB + Index: 0.05MB
wpqj_postmeta: Data: 0.38MB + Index: 0.25MB
wpqj_posts: Data: 0.22MB + Index: 0.06MB
wpqj_termmeta: Data: 0.02MB + Index: 0.03MB
wpqj_terms: Data: 0.02MB + Index: 0.03MB
wpqj_term_relationships: Data: 0.02MB + Index: 0.02MB
wpqj_term_taxonomy: Data: 0.02MB + Index: 0.03MB
wpqj_usermeta: Data: 0.02MB + Index: 0.03MB
wpqj_users: Data: 0.02MB + Index: 0.05MB

### Post Type Counts ###

attachment: 150
jetpack_migration: 2
jp_img_sitemap: 1
jp_sitemap: 1
jp_sitemap_master: 1
page: 6
post: 2
product: 139
revision: 9

### Security ###

Secure connection (HTTPS): ❌Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (16) ###

Admin Menu Editor: by Janis Elsts – 1.8.1
Coming Soon Page & Maintenance Mode by SeedProd: by SeedProd – 5.0.15
Hello Dolly: by Matt Mullenweg – 1.6
Jetpack by WordPress.com: by Automattic – 5.5
Menu Items Visibility Control: by Hassan Derakhshandeh – 0.3.5
Minimal Coming Soon & Maintenance Mode: by WebFactory – 1.35
PW WooCommerce Bulk Edit: by Pimwick
LLC – 2.34 – Not tested with the active version of WooCommerce

Storefront Header Picker: by Wooassist – 1.0.2
Storefront Product Sharing: by WooThemes – 1.0.4
Title Toggle for Storefront Theme: by Wooassist – 1.2.4
UpdraftPlus - Backup/Restore: by UpdraftPlus.Com
DavidAnderson – 1.13.12

WooCommerce Cart Tab: by jameskoster – 1.1.1 – Not tested with the active version of WooCommerce
WooCommerce PayPal Powered by Braintree Gateway: by WooCommerce – 2.0.4 – Not tested with the active version of WooCommerce
WooCommerce Stripe Gateway: by WooCommerce – 3.2.3 – Not tested with the active version of WooCommerce
WooCommerce Services: by Automattic – 1.8.3 – Not tested with the active version of WooCommerce
WooCommerce: by Automattic – 3.2.3

### Settings ###

API Enabled: ✔
Force SSL: ✔
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

### WC Pages ###

Shop base: #4 - /shop/
Cart: #5 - /cart/
Checkout: #6 - /checkout/
My account: #7 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 2.2.5
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –
`
</details>

**Comment by mikejolley** [comment]
The exporter/importer does not facilitate drafts. The exporter exports private and published products, not drafts (which are most likely incomplete). As seen:

https://github.com/woocommerce/woocommerce/blob/master/includes/export/class-wc-product-csv-exporter.php#L129

This is intentional. There is a filter the line above where I linked should you wish to customise this.

Thanks

**Comment by fbjohn** [comment]
Then the drop down shouldn’t say “all products”.
Thanks for showing me how to customize it on my end. 

It should be clear to the end user that they are not exporting all products. I have over 100 products but only 3 published because that’s the only ones I was able to finish. I need to export the rest of them so that I can bulk edit them in a spreadsheet. (So that I can finish them.). 

If I had just the opposite and had over 100 published and only 3 drafts and I click the export all products and all columns, I would be assuming those 3 were in there. I’d be really upset if I found out those 3 got left out. 

So it’s either still a bugs, or at the very least, very deceptive to the end user. 

Thanks,

John Moore
